contracts/FungibleToken.cdc

```diff
@@ -182,6 +182,12 @@ pub contract interface FungibleToken {
         /// deposit takes a Vault and adds its balance to the balance of this Vault
         ///
         pub fun deposit(from: @Vault) {
+            // Assert that the concrete type of the deposited vault is the same
+            // as the vault that is accepting the deposit
+            pre {
+                from.getType().identifier.slice(from: 0, upTo: 18) == self.getType().identifier.slice(from: 0, upTo: 18): 
+                    "Cannot deposit an incompatible token type"
```

Comment:
This code does not match the comment, the code checks that the types are declared in the same account. Was that what you wanted to ensure?

If you really want to check that the type of the vault passes as an argument matches that of the vault that is accepting the deposit, you can just use `self.isInstance` and `from.getType()`:

```suggestion
                from.isInstance(self.getType()): 
                    "Cannot deposit an incompatible token type"
```

I've added a test case for this in onflow/cadence#779, see https://github.com/onflow/cadence/commit/4f93166492e3cdfb3c063c39a60d81d29570f9f9

It should also be possible to just compare the types: `from.getType() == self.getType()`

Reply:
oh yeah that makes a lot more sense! I'll change it to that. Thanks for the suggestion!

Reply:
Fixed in c84d060